### PR TITLE
Add getTableTypes extension point to BaseJdbcClient

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -774,7 +774,12 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
                 connection.getCatalog(),
                 escapeNamePattern(schemaName, metadata.getSearchStringEscape()).orElse(null),
                 escapeNamePattern(tableName, metadata.getSearchStringEscape()).orElse(null),
-                new String[] {"TABLE", "VIEW"});
+                getTableTypes().toArray(String[]::new));
```
Comment (findepi) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java:777]: Do we want to allow subclasses to provide `null` here (perhaps as `Optional<List<String>>`)?

**Review by findepi — APPROVED** [review]
Mind CI

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java:777]: Good point.

Comment (losipiuk) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java]: ac

**Review by findepi — APPROVED** [review]

**Pull request merged**